david_recur2.py

```diff
@@ -0,0 +1,19 @@
+# function returning all permutations of chars of a given string in alphabetical order
+
+def find_all_permutations (input_string):   
+    all_permutations = []
+    input_string = "".join(sorted(input_string))
```

Comment:
because i want the permutations in alphabetical order?
it's either sort the input or sort the output, and the size of the input is significantly smaller than that of the output?